spec/request/user/create_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'POST #create sign_up', type: :request do  
+  let!(:user_params) { { user: FactoryBot.attributes_for(:user) } }
+  subject do
+    post '/api/v1/users', params: user_params, as: :json
+  end
+
+  context 'when valid' do
+    it 'does get a sucessful answer' do
+      subject
+      expect(response).to be_successful
+    end
+
+    it 'does create a new user' do
+      expect { subject }.to change { User.count }.by(1)
+    end
+
+    it 'the attributes from params persist on db' do
+      subject
+      expect(response.body).to include_json(
+        user: {
+          email: User.last.email,
+          gender: User.last.gender,
+          full_name: User.last.full_name
+        }
+      )
+    end
+  end
+
+  context 'when invalid' do
+    context 'when missing argument' do
+      it '(email) it does not create a new user' do
+        user_params[:user][:email] = nil
+        subject
+        expect(response).not_to be_successful
```

Comment:
Maybe here and in the tests below you could check for a more specific error status. And also you could check for the error message. Like:

```
context '(email) it does not create a new user' do
  before do
    user_params[:user][:email] = nil
    subject
  end

  it 'returns unauthorized status' do
    expect(response).to have_http_status(:unauthorized)
  end

  it 'returns error message' do
    expect(json[:error]).to eq('some error message')
  end
end
```